src/librustc_errors/lib.rs

```diff
@@ -23,16 +23,9 @@
 #![feature(staged_api)]
 #![feature(range_contains)]
 #![feature(libc)]
-#![feature(unicode)]
 
-extern crate serialize;
 extern crate term;
-#[macro_use]
-extern crate log;
-#[macro_use]
 extern crate libc;
-extern crate std_unicode;
-extern crate serialize as rustc_serialize; // used by deriving
```

Comment:
It's true yeah that it needs to be propagated outwards, but this may not need rustc -> cargo support. The compiler could simply warn about unused `--extern` flags, and we could perhaps tailor it so the warning makes sense if you're just using Cargo.

In any case that seems like future work to me.